core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceAclManager.scala

```diff
@@ -102,4 +111,47 @@ class MultiCloudWorkspaceAclManager(workspaceManagerDAO: WorkspaceManagerDAO, va
         )
     }
   )
+
+  def maybeShareWorkspaceNamespaceCompute(
+    policyAdditions: Set[(SamResourcePolicyName, String)],
+    workspaceName: WorkspaceName,
+    ctx: RawlsRequestContext
+  ): Future[Unit] = {
+    val newWriterEmails = policyAdditions.collect { case (SamWorkspacePolicyNames.writer, email) => email }
+
+    if (newWriterEmails.nonEmpty) {
```

Comment:
A suggestion for an alternative structure:
```scala
  lazy val workspaceBillingProfileId: Future[UUID] = dataSource.inTransaction(
          _.rawlsBillingProjectQuery.load(RawlsBillingProjectName(workspaceName.namespace)
    )
    .map(_.getOrElse(throw new RawlsExceptionWithErrorReport(/*...*/))
    .billingProfileId
    .map(UUID.fromString(_))
    .getOrElse(throw new RawlsExceptionWithErrorReport(/*...*/))
    
  Future.traverse(newWriterEmails) { email =>
      Future(
          billingProfileManagerDAO.addProfilePolicyMember(
              UUID.fromString(workspaceBillingProfileId),
              ProfilePolicy.PetCreator,
              email,
              ctx
          )
        )
  }
```
This avoids the if with the `else Future.successful()` all the way at the end (which I found a little hard to follow at first glance), since the loop only executes if there are elements in newWriterEmails, and the lazy value is only evaluated if the loop executes at least once.

Reply:
I swapped the conditional to `.isEmpty` so the `Future.successful()` will be at the top and make it clearer that it will short circuit